classes/task/process_duplicated_event_module_fix.php

```diff
@@ -0,0 +1,149 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Adhoc task to check and fix episode modules in the newly imported course.
+ *
+ * @package     block_opencast
+ * @copyright   2024 Farbod Zamani Boroujeni, ELAN e.V.
+ * @author      Farbod Zamani Boroujeni <zamani@elan-ev.de>
+ * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace block_opencast\task;
+
+use block_opencast\local\apibridge;
+use block_opencast\local\notifications;
+use block_opencast\local\importvideosmanager;
+use core\task\adhoc_task;
+use Exception;
+use moodle_exception;
+
+/**
+ * Adhoc task to check and fix episode modules in the newly imported course.
+ *
+ * @package     block_opencast
+ * @copyright   2024 Farbod Zamani Boroujeni, ELAN e.V.
+ * @author      Farbod Zamani Boroujeni <zamani@elan-ev.de>
+ * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class process_duplicated_event_module_fix extends adhoc_task {
```

Comment:
Could you please add your suggestion!
And is it something that older versiuon Moodle that the plugin supports could handle as well?

Reply:
I added it, but have not yet tested it, because I don't have my test instance setup here. But it should work, and I will definitely test it once more before merging :)